python/ml4ir/base/features/feature_fns/categorical.py

```diff
@@ -220,6 +220,33 @@ def categorical_embedding_with_vocabulary_file(feature_tensor, feature_info, fil
     return embedding
 
 
+class CategoricalDropout(layers.Layer):
+    """Custom Dropout class for categorical indices"""
+
+    def __init__(self, dropout_rate, seed=None, **kwargs):
+        """
+        Args:
+            dropout_rate: fraction of units to drop, i.e., set to OOV token 0
+            seed: random seed for sampling
+        """
+        super(CategoricalDropout, self).__init__(**kwargs)
+        self.dropout_rate = dropout_rate
+        self.seed = seed
+
+    def call(self, inputs, training=None):
```

Comment:
No unnecessary scaling and de-scaling anymore. Just a simple mask. Can even rename to `CategoricalMask` to avoid confusion in the future.